### PR TITLE
support kryo decorators

### PR DESCRIPTION
**Pull request by jasonjckn**

### CODE REVIEW

File: conf/storm.yaml.example [hunk 1]
```diff
@@ -13,6 +13,10 @@
 #     - org.mycompany.MyType
 #     - org.mycompany.MyType2: org.mycompany.MyType2Serializer
 #
+## List of custom kryo decorators
+# topology.kryo.register:
```
Comment (nathanmarz) [inline conf/storm.yaml.example]: typo here?

File: src/jvm/backtype/storm/Config.java [hunk 2]
```diff
@@ -373,6 +373,14 @@
     public static String TOPOLOGY_KRYO_REGISTER = "topology.kryo.register";
 
     /**
+     * A list of classes that customize storm's kryo instance during start-up.
```
Comment (nathanmarz) [inline src/jvm/backtype/storm/Config.java:376]: Let's make sure to fill in the Javadoc here

File: src/jvm/backtype/storm/serialization/SerializationFactory.java [hunk 3]
```diff
@@ -88,6 +88,30 @@ public static Kryo getKryo(Map conf) {
             }
         }
         k.overrideDefault(true);
+
+        if (!(conf.get(Config.TOPOLOGY_KRYO_DECORATORS) instanceof List)) {
```
Comment (nathanmarz) [inline src/jvm/backtype/storm/serialization/SerializationFactory.java]: Why the instanceof check? Shouldn't we just check for null?

**Comment by nathanmarz** [comment]
One more thing is necessary in this pull request, which is to normalize the configuration for this property. See: https://github.com/nathanmarz/storm/blob/master/src/clj/backtype/storm/daemon/nimbus.clj#L768

This is important to guarantee that every task have the same serialization registrations – like topology.kryo.register, you need to merge in all the component-specific decorators with the topology decorators, making sure to unique the list of decorator classes.

File: src/jvm/backtype/storm/serialization/SerializationFactory.java [hunk 4]
```diff
@@ -88,6 +88,30 @@ public static Kryo getKryo(Map conf) {
...
+            return k;
+        }
+
+        for(String klassName : (List<String>)conf.get(Config.TOPOLOGY_KRYO_DECORATORS)) {
+            try {
+                Class klass = Class.forName(klassName);
+                IKryoDecorator decorator = (IKryoDecorator)klass.newInstance();
+                decorator.decorate(k);
+                System.out.println(klassName + ".. executed decorate.");
```
Comment (glennmcallister) [inline src/jvm/backtype/storm/serialization/SerializationFactory.java]: Was this intended to be a long term debugging log or just a work-in-progress message for yourself?

Comment (jasonjckn) [inline src/jvm/backtype/storm/serialization/SerializationFactory.java]: meant to remove that, thanks!

**Pull request merged**